.github/workflows/build.yml

```diff
@@ -48,28 +62,46 @@ jobs:
 
   build:
     needs: build-sentry-native
-    name: .NET (${{ matrix.os }})
-    runs-on: ${{ matrix.os }}
+    name: .NET (${{ matrix.runner.container || matrix.runner.os }})
+    runs-on: ${{ matrix.runner.os }}
+    container: ${{ matrix.runner.container }}
 
     strategy:
       fail-fast: false
       matrix:
         # Pin ubuntu to ensure mono is installed
         # Pin macos to get the version of Xcode that we need: https://github.com/actions/runner-images/issues/10703
-        os: [ubuntu-22.04, windows-latest, macos-15]
+        runner: [
+          {os: ubuntu-22.04},
+          {os: ubuntu-latest, container: alpine:edge, target: Linux-musl},
+          {os: windows-latest},
+          {os: macos-15}
+        ]
 
     steps:
       - name: Cancel Previous Runs
         if: github.ref_name != 'main' && !startsWith(github.ref_name, 'release/')
         uses: styfle/cancel-workflow-action@85880fa0301c86cca9da44039ee3bb12d3bedbfa # Tag: 0.12.1
 
+      - name: Install build dependencies for Alpine Linux
+        if: ${{ startsWith(matrix.runner.container, 'alpine') }}
+        run: |
+          apk update
+          apk add bash build-base cmake curl-dev git icu mono openjdk11 powershell sudo tar wget
+          git config --global --add safe.directory $GITHUB_WORKSPACE
+          # Grpc.Tools protoc is prebuilt with glibc (https://github.com/grpc/grpc/issues/24188#issuecomment-1403435551)
+          apk add grpc-plugins
+          echo "PROTOBUF_PROTOC=/usr/bin/protoc" >> $GITHUB_ENV
+          echo "GRPC_PROTOC_PLUGIN=/usr/bin/grpc_csharp_plugin" >> $GITHUB_ENV
```

Comment:
Using the edge branch is probably a bad idea, as it may suddenly change, but the [mono](https://pkgs.alpinelinux.org/packages?name=mono&branch=edge) is currently only available in the edge branch, so I'm experimenting with that for now. We could consider building a custom Docker image to hide all this boilerplate...